### PR TITLE
[Diagnostics] SR-7349 Improve accessibility diagnostics for inheritance from generic classes

### PR DESCRIPTION
**Pull request by AnthonyLatsis**
Resolves https://bugs.swift.org/browse/SR-7349

Have updated only the warning, going to update the error version of the diagnostic in a few hours.

### CODE REVIEW

**Review by xedin — APPROVED** [review]
LGTM, thank you!

**Comment by xedin** [comment]
@swift-ci please smoke test and merge

**Comment by xedin** [comment]
I thought that this is what you message meant :)

**Comment by AnthonyLatsis** [comment]
Alright, no problem. I can request to merge the same branch, right?

**Comment by xedin** [comment]
I think that should be okay, but I managed to stop the 'test + merge' job so we can have it all in one. Please add `[WIP]` or `[Don't Merge]` next time so it's more obvious that you don't want to split the changes...

**Comment by xedin** [comment]
@swift-ci please test

**Comment by xedin** [comment]
@swift-ci please test

**Comment by xedin** [comment]
Merged, thank you!

File: lib/Sema/TypeCheckDecl.cpp [hunk 1]
```diff
@@ -1785,51 +1785,41 @@ static void checkGenericParamAccess(TypeChecker &TC,
     return;
 
   // This must stay in sync with diag::generic_param_access.
-  enum {
-    ACEK_Parameter = 0,
-    ACEK_Requirement
+  enum ACEK {
+    Parameter = 0,
+    Requirement
```
Comment (jrose-apple) [inline lib/Sema/TypeCheckDecl.cpp]: This pollutes the surrounding namespace unless the enum is declared with `enum class`.

**Review by jrose-apple — COMMENTED** [review]
Thanks, Anthony!

File: lib/Sema/TypeCheckDecl.cpp (hunk 1 above)
Comment (AnthonyLatsis) [reply lib/Sema/TypeCheckDecl.cpp]: Ah, true. It is OK to commit the change here?

Comment (xedin) [reply lib/Sema/TypeCheckDecl.cpp]: You can open separate PR for to correct this, it's not a problem!

**Pull request merged**